addons/cert.py

```diff
@@ -234,6 +245,101 @@ def msc30(data):
         if simpleMatch(token, "rand ( )") and isStandardFunction(token):
             reportError(token, 'style', 'Do not use the rand() function for generating pseudorandom numbers', 'MSC30-c')
 
+# STR03-C
+# Do not inadvertently truncate a string
+def str03(data):
+    for token in data.tokenlist:
+        if token.str=='strncpy':
+            nextToken = token.astParent
+            while(nextToken):
+                if nextToken.str=='sizeof':
+                    reportError(nextToken, 'style', 'Do not inadvertently truncate a string', 'STR03-C')
+                nextToken = nextToken.next
+
+# STR05-C
+# Use pointers to const when referring to string literals
+def str05(data):
+    for token in data.tokenlist:
+        if token.isString:
+            parent = token.astParent
+            if parent is None:
+                continue
+            parentOp1 = parent.astOperand1
```

Comment:
You should probably check that parentOp1 is not None.. it is valid syntax to just write a string literal as far as I know.